test/e2e/prometheus_test.go

```diff
@@ -92,3 +97,180 @@ func TestPrometheusAlertmanagerAntiAffinity(t *testing.T) {
+      - name: serving-certs-ca-bundle
+        configMap:
+	  name: serving-certs-ca-bundle
+      - name: certs
+        emptyDir: {}
+      initContainers:
+      - command:
+        - /bin/sh
+        - -c
+        - 'cp /etc/certs/tls/tls.key /etc/certs/key.pem; cat /etc/certs/tls/tls.crt > /etc/certs/cert.pem; cat /etc/certs/ca/service-ca.crt >> /etc/certs/cert.pem'
+	name: crypt-init
+	image: busybox
+	volumeMounts:
+	- mountPath: /etc/certs
+	  name: certs
+        - mountPath: /etc/certs/tls
+          name: rw-target-tls
+        - mountPath: /etc/certs/ca
+          name: serving-certs-ca-bundle
+`
+	deployment := &appsv1.Deployment{}
+	if err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(targetDeployment)), 100).Decode(&deployment); err != nil {
+		t.Fatal(err)
+	}
+
+	// create service with annotation for service-ca operator
+	name := "test"
+	svc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"group": name,
+			},
+			Namespace: f.Ns,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/serving-cert-secret-name": "rw-target-tls",
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Type: v1.ServiceTypeLoadBalancer,
+			Ports: []v1.ServicePort{
+				{
+					Name: "web",
+					Port: 8080,
+				},
+				{
+					Name: "mtls",
+					Port: 8081,
+				},
+			},
+			Selector: map[string]string{
+				"group": name,
+			},
+		},
+	}
+
+	if err := f.OperatorClient.CreateOrUpdateService(ctx, svc); err != nil {
+		t.Fatal(err)
+	}
+	// deploy remote write target
+	if err := f.OperatorClient.CreateOrUpdateDeployment(ctx, deployment); err != nil {
+		t.Fatal(err)
+	}
+	svc, err := f.KubeClient.CoreV1().Services(f.Ns).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, scenario := range []struct {
+		name  string
+		proto string
+	}{
+		// check remote write logs
+		{
+			name:  "assert remote write to http works",
+			proto: "http",
+		},
+	} {
+		url := scenario.proto + "://" + svc.Spec.ClusterIP + ":" + fmt.Sprint(svc.Spec.Ports[1].Port)
+
+		data := fmt.Sprintf(`prometheusK8s:
+  logLevel: debug
+  retention: 10h
+  tolerations:
+    - operator: "Exists"
+  remoteWrite:
+  - url: "%s"
+`, url)
+
+		if err := f.OperatorClient.CreateOrUpdateConfigMap(ctx, configMapWithData(t, data)); err != nil {
+			t.Fatal(err)
+		}
+
+		promLogs0, err := f.GetLogs(f.Ns, "prometheus-k8s-0", "prometheus")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		promLogs1, err := f.GetLogs(f.Ns, "prometheus-k8s-1", "prometheus")
+		if err != nil {
+			t.Fatal(err)
+		}
+		var promLogs strings.Builder
+		promLogErr := "prometheus logs are empty"
+		if i, _ := promLogs.WriteString(promLogs0); i == 0 {
+			t.Fatal(promLogErr)
+		}
+		if i, _ := promLogs.WriteString(promLogs1); i == 0 {
+			t.Fatal(promLogErr)
+		}
+
+		rwEndpointOpts := metav1.ListOptions{
+			LabelSelector: fields.SelectorFromSet(fields.Set(map[string]string{
+				"group": name,
+			})).String(),
```

Comment:
This can be simplified by using [labels.FormatLabels](https://pkg.go.dev/k8s.io/apimachinery/pkg/labels#FormatLabels).